### PR TITLE
Adds 'off' icons for nanite machinery

### PR DESCRIPTION
**Pull request by XDTM**
## About The Pull Request

Adds 'off' icons to the nanite program hub, the nanite programmer, and the nanite cloud console.

To be exact, it converts the screen part into an overlay, which has the additional benefit of using the same visible-through-darkness effect that consoles use.
Also added a broken overlay for the nanite cloud console, which features a cracked screen.

## Why It's Good For The Game

I spent way too much time trying to find out why i could no longer interact with nanite machinery while debugging before i noticed that the area power had just ran out

## Changelog
:cl: XDTM
imageadd: Nanite machinery's screens now turn off if they're broken or are unpowered, and are visible through darkness when on.
/:cl:

### CODE REVIEW

**Comment by MrHorizons** [comment]
Why not put the blinky lights on the cloud controller in with the screen overlay?

**Comment by GunHog** [comment]
The programmer and clould controller sprites seem to have a stray pixel.

**Comment by XDTM** [comment]
> 
> 
> Why not put the blinky lights on the cloud controller in with the screen overlay?

Mostly because i really didn't want to spend the time to cut them out of each of the 32 animation frames after i noticed i had missed them

File: code/modules/research/nanites/nanite_program_hub.dm [hunk 1]
```diff
@@ -26,6 +26,14 @@
 	. = ..()
 	linked_techweb = SSresearch.science_tech
 
+/obj/machinery/nanite_program_hub/update_icon()
+	. = ..()
+	SSvis_overlays.remove_vis_overlay(src, managed_vis_overlays)
+	if((machine_stat & (NOPOWER|MAINT|BROKEN)) || panel_open)
+		return
+	SSvis_overlays.add_vis_overlay(src, icon, "nanite_program_hub_on", layer, plane)
+	SSvis_overlays.add_vis_overlay(src, icon, "nanite_program_hub_on", EMISSIVE_LAYER, EMISSIVE_PLANE)
```
Comment (ShizCalev) [inline code/modules/research/nanites/nanite_program_hub.dm:35]: why aren't you doing this via update_overlays?

File: code/modules/research/nanites/nanite_programmer.dm [hunk 2]
```diff
@@ -11,6 +11,14 @@
 	flags_1 = HEAR_1
 	circuit = /obj/item/circuitboard/machine/nanite_programmer
 
+/obj/machinery/nanite_programmer/update_icon()
+	. = ..()
+	SSvis_overlays.remove_vis_overlay(src, managed_vis_overlays)
+	if((machine_stat & (NOPOWER|MAINT|BROKEN)) || panel_open)
+		return
+	SSvis_overlays.add_vis_overlay(src, icon, "nanite_programmer_on", layer, plane)
+	SSvis_overlays.add_vis_overlay(src, icon, "nanite_programmer_on", EMISSIVE_LAYER, EMISSIVE_PLANE)
```
Comment (ShizCalev) [inline code/modules/research/nanites/nanite_programmer.dm:20]: why aren't you doing this via update_overlays?

**Review by ShizCalev — CHANGES_REQUESTED** [review]

File: code/modules/research/nanites/nanite_program_hub.dm (hunk 1 above)
Comment (XDTM) [reply code/modules/research/nanites/nanite_program_hub.dm:35]: my bad, i initially did, it didn't work, changed it to update_icons to try to fix it, and forgot to switch it back after i found the actual issue

**Review by Twaticus — APPROVED** [review]

**Pull request merged**